client/core/trade.go

```diff
@@ -281,12 +282,13 @@ func (t *trackedTrade) nomatch(oid order.OrderID) error {
 		t.notify(newOrderNote("Order booked", "", db.Data, corder))
 	} else {
 		t.returnCoins()
+		assets.count(t.wallets.fromAsset.ID)
```

Comment:
The asset map return from `nomatch` is a bit peculiar since it can never be more than one asset in the map, but I see how it simplifies `handleNoMatchRoute` a bit with (`if len(updatedAssets) > 0`).  An alt would be to return a `*uint32` an do `c.updateAssetBalance(*updatedAsset)` if the return is non-ni.  That's a bit contorted too though.  Can leave.